apps/remix-ide/src/app/tabs/runTab/settings.js

```diff
@@ -53,6 +53,28 @@ class SettingsUI {
     })
   }
 
+  validateInputKey (e) {
+    // preventing not numeric keys
+    // preventing 000 case
+    if (!helper.isNumeric(e.key) ||
+      (e.key === '0' && !parseInt(this.el.querySelector('#value').value) && this.el.querySelector('#value').value.length > 0)) {
```

Comment:
Is e.key === '0' consistent for all browsers? You can also have this (`this.el.querySelector('#value')` ) in a variable to make it more readable.

Reply:
it is